tests/integration/query/simple/with_multiple_types_test.go

```diff
@@ -0,0 +1,323 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package simple
+
+import (
+	"testing"
+
+	testUtils "github.com/sourcenetwork/defradb/tests/integration"
+)
+
+func TestSimple_WithSevenDummyTypesBefore(t *testing.T) {
+	test := testUtils.TestCase{
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+					type Type0 {
+						f: String
+					}
+					type Type1 {
+						f: String
+					}
+					type Type2 {
+						f: String
+					}
+					type Type3 {
+						f: String
+					}
+					type Type4 {
+						f: String
+					}
+					type Type5 {
+						f: String
+					}
+					type Type6 {
+						f: String
+					}
+
+					type User {
+						name: String
+					}
```

Comment:
suggestion: for these tests would be nice to have some more information (as a documentation or test description) explaining why these weird-looking tests are necessary and which edge-case they test.

Reply:
Agreed, I'll add some in - thanks for pushing past my laziness here :)

- [ ] Add test docs